components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -404,7 +406,7 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
      * gitpod.io Extension point for implementing eligibility checks. Throws a ResponseError if not eligible.
      */
     protected async maySetTimeout(user: User): Promise<boolean> {
-        return this.eligibilityService.maySetTimeout(user);
+        return this.entitlementService.maySetTimeout(user, new Date());
```

Comment:
looking at `maySetTimeout`, the date param seems to be weird. it's used internally to express "until now", thus `new Date()`. but it doesn't seem to make any sense for `maySetTimeout`, does it? e.g. `maySetTimeout(user, yesterday)` is irrelevant. let's remove it, wdyt?

Reply:
> but it doesn't seem to make any sense for maySetTimeout, does it? 

:information_source:  It's mainly there for testability.

But I agree, it should have a default to not bother clients. Will change in a follow-up! :+1: